src/components/InventoryTable/InventoryTable.js

```diff
@@ -1,58 +1,26 @@
 /* eslint-disable react/display-name */
-/* eslint-disable camelcase */
-import React, {
-  Fragment,
-  forwardRef,
-  useEffect,
-  useRef,
-  useState,
-} from 'react';
+import React, { Fragment, forwardRef, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import { shallowEqual, useDispatch, useSelector, useStore } from 'react-redux';
+import { shallowEqual, useDispatch, useSelector } from 'react-redux';
+import debounce from 'lodash/debounce';
 import EntityTableToolbar from './EntityTableToolbar';
 import { TableToolbar } from '@redhat-cloud-services/frontend-components/TableToolbar';
 import { ErrorState } from '@redhat-cloud-services/frontend-components/ErrorState';
 import InventoryList from './InventoryList';
 import Pagination from './Pagination';
 import AccessDenied from '../../Utilities/AccessDenied';
 import { loadSystems } from '../../Utilities/sharedFunctions';
-import isEqual from 'lodash/isEqual';
 import { clearErrors, entitiesLoading } from '../../store/actions';
-import cloneDeep from 'lodash/cloneDeep';
 import { useSearchParams } from 'react-router-dom';
 import { ACTION_TYPES } from '../../store/action-types';
 
-/**
- * A helper function to store props and to always return the latest state.
- * For example, EntityTableToolbar wraps OnRefreshData in a callback, so we need this
- * to get the latest props and not the props at the time of when the function is
- * being wrapped in callback.
- */
-export const inventoryCache = () => {
```

Comment:
The inventoryCache seems unnecessary, since we hold these params and props in many places already.